utils/misc/misc.go

```diff
@@ -1394,3 +1393,17 @@ func ReverseInt(s []int) []int {
 func IsMultiTenant() bool {
 	return config.GetBool("EnableMultitenancy", false)
 }
+
+// lookup map recursively and return value
+func MapLookup(mapToLookup map[string]interface{}, defaultVal interface{}, keys ...string) interface{} {
+	if len(keys) == 0 {
+		return defaultVal
+	}
+	if val, ok := mapToLookup[keys[0]]; ok {
+		if len(keys) == 1 {
+			return val
+		}
+		return MapLookup(val.(map[string]interface{}), defaultVal, keys[1:]...)
```

Comment:
A type assertion would be valuable
```suggestion
		nextMap, ok := val.(map[string]interface{})
		if !ok {
			return defaultVal
		}
		return MapLookup(nextMap, defaultVal, keys[1:]...)
```

Reply:
Nice. Done.
This function brought to you by github copilot!